desktop-widgets/mainwindow.cpp

```diff
@@ -77,8 +77,10 @@
     #include "plugins/facebook/facebookconnectwidget.h"
 #endif
 
-QProgressDialog *progressDialog = NULL;
-bool progressDialogCanceled = false;
+namespace {
+    QProgressDialog *progressDialog = nullptr;
+    bool progressDialogCanceled = false;
+}
```

Comment:
how does this help us?
Also, this commit includes `libdivecomputer` updates, hard NAK on that.

Reply:
The anonymous namespace is used to make things local to the compilation-unit. I.e. it has the same apparent effect as `static`.

Why did they introduce this strange thing, when there is `static` you may ask. Well, as Linus noted with quite some disdain, you can't declare member-functions or even classes as `static` [technically you can - but it has an entirely different meaning]. Therefore, if you don't want to export symbols of member functions, you have to use the anonymous namespace.

Whether C++ is silly for introducing this kludge and/or C is silly for using `static` for two different meanings, I'll leave to you to decide.